vendor/github.com/OpenBazaar/go-ethwallet/wallet/wallet.go

```diff
@@ -388,7 +389,25 @@ func (wallet *EthereumWallet) NewAddress(purpose wi.KeyPurpose) btcutil.Address
 
 // DecodeAddress - Parse the address string and return an address interface
 func (wallet *EthereumWallet) DecodeAddress(addr string) (btcutil.Address, error) {
-	ethAddr := common.HexToAddress(addr)
+	var ethAddr common.Address
+	if len(addr) > 64 {
```

Comment:
It would be nice to document the reason for this check somehow. Either with a variable or a comment.

Reply:
Would also be nice to see a test case protecting both versions of Decoding inputs.

Reply:
I extracted the ethScript conversion to document this a bit. Still missing a unit test though.